service/history/workflow/context.go

```diff
@@ -178,10 +178,15 @@ func NewContext(
 	throttledLogger log.ThrottledLogger,
 	metricsHandler metrics.Handler,
 ) *ContextImpl {
+	tags := []tag.Tag{
+		tag.WorkflowNamespaceID(workflowKey.NamespaceID),
+		tag.WorkflowID(workflowKey.WorkflowID),
+		tag.WorkflowRunID(workflowKey.RunID),
+	}
 	return &ContextImpl{
 		workflowKey:     workflowKey,
-		logger:          logger,
-		throttledLogger: throttledLogger,
+		logger:          log.With(logger, tags...),
```

Comment:
This logger is passed into the new Update registry here: https://github.com/temporalio/temporal/pull/6299/files#diff-c365166a3b00b43950b46bc9be819b9c407f41947377a563c8b18369bd5d897fR911